plugins/test/unit/plugins/distributors/test_steps.py

```diff
@@ -86,6 +89,28 @@ def test_process_main(self, lib, add_ref):
         lib.Summary.assert_called_once_with(repository)
         lib.Summary.return_value.generate.assert_called_once_with()
 
+    @patch(MODULE + '.lib')
+    def test_process_main_exception(self, lib):
+        units = [
+            Mock(branch='branch:1', commit='commit:1', storage_path='path:1'),
+        ]
+        repository = Mock()
+        repository.pull_local.side_effect = LibError
+        lib.Repository.return_value = repository
+        lib.LibError = LibError
+        parent = Mock(publish_dir='/tmp/dir-1234', config={})
+
+        # test
+        main = steps.MainStep()
+        main._get_units = Mock(return_value=units)
+        main.parent = parent
+
+        try:
+            main.process_main()
+            self.assertTrue(False, msg='Publish exception expected')
```

Comment:
I like to use ``assertRaises`` as a context manager, which lets you inspect the raised exception while avoiding this kind of thing. It's roughly:

```
with self.assertRaises(PulpCodedException) as assertion:
  main.process_main()

self.assertEqual(assertion.exception.error_code, errors.OST0006)
```